tests/Integration/Tracking/GlobalSiteTagTest.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+ * Integration tests for the GlobalSiteTag class.
+ *
+ * This test suite validates that the GlobalSiteTag hooks register correctly
+ * and that key pixel events can be triggered on the frontend.
+ *
+ * @package SnapchatForWooCommerce\Tests\Integration\Tracking
+ */
+
+namespace SnapchatForWooCommerce\Tests\Integration\Tracking;
+
+use WP_UnitTestCase;
+use SnapchatForWooCommerce\Tracking\GlobalSiteTag;
+
+/**
+ * @covers \SnapchatForWooCommerce\Tracking\GlobalSiteTag
+ */
+class GlobalSiteTagTest extends WP_UnitTestCase {
+
+	/**
+	 * Service under test.
+	 *
+	 * @var GlobalSiteTag
+	 */
+	private $service;
+
+	public function set_up(): void {
+		parent::set_up();
+
+		$this->service = new GlobalSiteTag();
+		$this->service->register();
+	}
+
+	/**
+	 * Tests that GlobalSiteTag hooks are registered.
+	 */
+	public function test_hooks_are_registered() {
+		$this->assertNotFalse( has_filter( 'woocommerce_loop_add_to_cart_link' ) );
+		$this->assertNotFalse( has_action( 'woocommerce_after_add_to_cart_button' ) );
+		$this->assertNotFalse( has_action( 'woocommerce_after_single_product' ) );
+		$this->assertNotFalse( has_action( 'woocommerce_before_thankyou' ) );
+		$this->assertNotFalse( has_action( 'wp_footer' ) );
+	}
+
+	/**
+	 * Tests that track_view_content_event() runs without error.
+	 *
+	 * Simulates a single product page view.
+	 */
+	public function test_track_view_content_event_runs_without_error() {
```

Comment:
It looks like this and the following tests are not yet fully implemented since `$this->assertTrue( true );` will always pass. Is that intentional?

Reply:
Yeah I initially just added sanity checks and thought I'd test the pixel script in E2E. I've added the complete unit tests here now.